packages/react-router/src/useMatch.tsx

```diff
@@ -13,8 +13,10 @@ export type UseMatchOptions<
   TStrict extends boolean,
   TRouteMatch,
   TSelected,
+  TThrow extends boolean,
 > = StrictOrFrom<TFrom, TStrict> & {
   select?: (match: TRouteMatch) => TSelected
+  shouldThrow?: TThrow
```

Comment:
I'd rather this be `throw` to match that API surface with what we are already doing with `redirect()`.